api-client/src/main/java/com/desk/java/apiclient/util/OpportunityActivityAdapter.java

```diff
@@ -106,6 +110,10 @@ public IOpportunityActivity deserialize(JsonElement jsonElement, Type type, Json
 
                         } else if (OpportunityActivityTypes.TASK.equalsIgnoreCase(activityType.getAsString())) {
                             return gson.fromJson(jsonElement, OpportunityTask.class);
+
```

Comment:
If the activity type field is changed to be the enum then we could just have a switch here to check the type.